src/Chainweb/Pact/RestAPI/Server.hs

```diff
@@ -154,25 +156,26 @@ listenHandler
     -> ChainResources logger
     -> TransactionBloomCache
     -> ListenerRequest
-    -> Handler ApiResult
+    -> Handler (CommandResult Hash)
 listenHandler cutR cid chain bloomCache (ListenerRequest key) =
-    liftIO $ handleTimeout runListen
+    liftIO (handleTimeout runListen) >>= handleResult
   where
-    nullResponse = ApiResult (object []) Nothing Nothing
+    handleResult (Right r) = return r
+    handleResult _ = throwError $ err400 { errBody = "Timeout" }
 
     runListen timedOut = go Nothing
       where
         go !prevCut = do
             m <- waitForNewCut prevCut
             case m of
-                Nothing -> return nullResponse      -- timeout
```

Comment:
@gregorycollins I think I have the semantics right here, please confirm

Reply:
Timeout is an expected condition here, it shouldn't return 400. Instead the CommandResult should be populated with a timeout code if possible. I guess it doesn't matter as long as the web client code does the right thing with respect to retry. If you do use an HTTP error code, properly it should be 500-series because 400-series responses are meant to indicate some problem with the request.

Reply:
What should it return then? 500 "Timeout"? cloudflare does a "522 Timeout" https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#Cloudflare

Reply:
If the timeout is expected. What about `204 No Content` or `202 Accepted`?

Reply:
I like `204`

Reply:
OTOH, maybe we should just do 200 with `{ "status": "timeout" }` body, just to avoid finicky header code docs. One thing I didn't do with this update is go for the "literate error responses", ie use `errorResponse` or whatever to make 402s etc have a parseable body with `{ "status": "failure", "code": "402", "message": ... }` as discussed, so that JS frontends can simply parse a body.